src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -448,10 +529,19 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
             // Render an empty JSON object instead of null. This allows a round trip where this property name is the
             // key to a map which would otherwise not be set and result in a different object.
             // This affects envelope size which isn't recomputed after a roundtrip.
-            if (originalPropertyDepth == writer.CurrentDepth)
```

Comment:
This was the actual cause of the original bug.  The assumption that the depth won't have change is true when a property being serialized throws an exception on its own, but is false when an exception occurs for reaching max depth.